### DISCUSSION TITLE
Advice please: Restarted node with old lnddir state (accident)

### DISCUSSION THREAD

**Discussion by dunkfan** [lightningnetwork/lnd#9019] [Q&A]
I made the unfortunate mistake of restarting a node with the wrong lnd directory. It was a backup copy which was a few days old. Not many routing transactions had been made but some channel changes had taken place. 
As expected that meant all channels were force closed and unfortunately I lost quite a few sats.
walletbalance is large but not where it should be :(
Currently my node is claiming all channels are gone (except for one private channel).

Notably, one of the new channels which was shut down does not show up in the "closedchannels" list.
Channel ID for that was: 942015383245488129 It was holding .1 BTC 
Some advice would be welcome on what steps I should take - if any - to attempt to retrieve the lost funds.
For starters, should I restart the node with the correct lnd directory where for example the new channel existed? Could that help in retrieving the liquidity in that particular channel?

**Comment by guggero** [comment]
Did you force close any channels manually? If not, then you should be somewhat safe.
The missing channel is probably newer than the backup was, it was from less than a week ago.

I think it would make sense to start the node again from the new directory. Then check the `lncli pendingchannels` output. Things might only move when transactions are confirmed, so some patience will be required.
But all sats (minus fees) should come back eventually (given you didn't manually force close any channels while running on the old version).

**Reply by dunkfan** [reply]
Sounds great!
No I havent done anything since, was too disappointed at my being fooled by the OS handling of a move directoy when a symbolic link was being used. I didnt even restart.... Will try that thanks

**Reply by dunkfan** [reply]
Didnt go very well. lnd crashes on startup trying to compact the database on what it should have originally run on
To make matters worse I recently went back from 0.18.2 to 0.17.3
Here's the console output.  
2024-08-20 21:13:54.863 [INF] LTND: Version: 0.17.3-beta commit=v0.17.3-beta, build=production, logging=default, debuglevel=info
2024-08-20 21:13:54.863 [INF] LTND: Active chain: Bitcoin (network=mainnet)
2024-08-20 21:13:54.910 [INF] RPCS: RPC server listening on 127.0.0.1:10009
2024-08-20 21:13:54.956 [INF] RPCS: gRPC proxy started at 127.0.0.1:8080
2024-08-20 21:13:54.957 [INF] LTND: Opening the main database, this might take a few minutes...
2024-08-20 21:13:54.957 [INF] LTND: Opening bbolt database, sync_freelist=false, auto_compact=true
2024-08-20 21:13:54.957 [INF] CHDB: Compacting database file at /home/pi/.lnd/data/graph/mainnet/channel.db
2024-08-20 21:14:08.645 [INF] LTND: Shutdown complete

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1063df8]

goroutine 1 [running]:
go.etcd.io/bbolt.(*Bucket).Sequence(...)
	go.etcd.io/bbolt@v1.3.7/bucket.go:338
github.com/lightningnetwork/lnd/kvdb.(*compacter).walkBucket.func1({0xed3fe0e1, 0xe, 0xe}, {0x0, 0x0, 0x0})
	github.com/lightningnetwork/lnd/kvdb@v1.4.4/bolt_compact.go:277 +0xdc
go.etcd.io/bbolt.(*Bucket).ForEach(0x32db1a0, 0x33b7128)
	go.etcd.io/bbolt@v1.3.7/bucket.go:389 +0x110
github.com/lightningnetwork/lnd/kvdb.(*compacter).walkBucket(0x33b75ec, 0x32db1a0, {0x3c00e80, 0x1, 0x1}, {0xdf61a0d2, 0xa, 0xa}, {0x0, 0x0, ...}, ...)
	github.com/lightningnetwork/lnd/kvdb@v1.4.4/bolt_compact.go:273 +0x17c
github.com/lightningnetwork/lnd/kvdb.(*compacter).walkBucket.func1({0xdf61a0d2, 0xa, 0xa}, {0x0, 0x0, 0x0})
	github.com/lightningnetwork/lnd/kvdb@v1.4.4/bolt_compact.go:276 +0x150
go.etcd.io/bbolt.(*Bucket).ForEach(0x3c06c80, 0x33b7208)
	go.etcd.io/bbolt@v1.3.7/bucket.go:389 +0x110
github.com/lightningnetwork/lnd/kvdb.(*compacter).walkBucket(0x33b75ec, 0x3c06c80, {0x0, 0x0, 0x0}, {0xedc1c41a, 0xa, 0xa}, {0x0, 0x0, ...}, ...)
	github.com/lightningnetwork/lnd/kvdb@v1.4.4/bolt_compact.go:273 +0x17c
github.com/lightningnetwork/lnd/kvdb.(*compacter).compact.(*compacter).walk.func3.1({0xedc1c41a, 0xa, 0xa}, 0x3c06c80)
	github.com/lightningnetwork/lnd/kvdb@v1.4.4/bolt_compact.go:226 +0x158
go.etcd.io/bbolt.(*Tx).ForEach.func1({0xedc1c41a, 0xa, 0xa}, {0x0, 0x0, 0x0})
	go.etcd.io/bbolt@v1.3.7/tx.go:130 +0x70
go.etcd.io/bbolt.(*Bucket).ForEach(0x318296c, 0x33b72e4)
	go.etcd.io/bbolt@v1.3.7/bucket.go:389 +0x110
go.etcd.io/bbolt.(*Tx).ForEach(0x3182960, 0x33b7304)
	go.etcd.io/bbolt@v1.3.7/tx.go:129 +0x58
github.com/lightningnetwork/lnd/kvdb.(*compacte

Can I start up and avoid the compact step?

**Reply by guggero** [reply]
You can turn off the compaction by setting `db.bolt.auto-compact=false` in your config.
But if compaction fails with that error, it means you have data corruption... So lnd will likely crash anyway, just later in the process... Seems like you're really being unlucky.

**Reply by dunkfan** [reply]
OK yes that failed quickly. I'm back up with the earlier version.
I have the channel.backup file with the correct channel state so perhaps using that would be the next unfortunate option?

**Reply by guggero** [reply]
Yes, using the channel.backup file should also allow you to sweep the closed channels (and ask the peer of the remaining open channel to also force close).

**Reply by dunkfan** [reply]
I used restorechanbackup with the incorrect but still working directory and channel.backup from the correct directory (Didnt use lncli create with my seed at all)
The .1 channel then closed leaving me a lot happier :)
There remains a private 'inactive' channel (not mine) where nothing moved for months with the liquidity on my side and 2 pending/waiting_close_channels. One of those has liquidity all on my side (zero transactions) and was known to both channel db files. Would using a force close here (the private inactive channel and the waiting ones) make any sense or should I just be patient?
Thanks so much

**Reply by guggero** [reply]
If you force close using an old state, that means you potentially could lose all funds if what you publish is an old state.
I would only do that if you are very certain the other party is never coming back (e.g. they shut down their node for good or also lost state). Unfortunately even then you can't be 100% sure they didn't use a watchtower.

So the safer route is to try to find out a way to contact the operators of those nodes and ask them to force close from their end.
